filebeat/tests/system/test_input.py

```diff
@@ -14,6 +15,73 @@
 
 
 class Test(BaseTest):
+    def test_fixup_registry_entries_with_global_id(self):
+        """
+        Should update the registry correctly when there were entries using the
+        default '.global' input ID
+        """
+        testfile = os.path.join(self.working_dir, "log", "test.log")
+        self.render_config_template(
+            template_name="filebeat-duplicated-id",
+            path=testfile,
+        )
+        os.mkdir(os.path.join(self.working_dir, "log/"))
+        file = open(testfile, 'w')
+        offset = 0  # keep count of the bytes written
+        for n in range(0, 5):
+            offset += file.write("hello world\n")
+        file.close()
+
+        os.makedirs(self.registry.path)
+        registry_file = os.path.join(self.registry.path, "log.json")
+
+        template_path = "./tests/system/input/registry-fix-global-id.j2"
+        self.render_template(template_path, registry_file, log_file=testfile, offset=offset)
+
+        shutil.copyfile(os.path.join(os.getcwd(), "tests", "system", "input", "registry-meta.json"),
+                        os.path.join(self.registry.path, "meta.json"))
+
+        proc = self.start_beat()
+
+        # wait for the "Start next scan" log message, this means the file has been
+        # fully harvested
+        self.wait_until(
+            lambda: self.log_contains(
+                "Start next scan"),
+            max_timeout=10)
+
+        proc.check_kill_and_wait()
+
+        reg = self.access_registry()
+        self.wait_until(reg.exists)
+        entries = [entry for entry in reg.load()]
+
+        # We got the latest entry for each key in the registry,
+        # they're ordered by the time of creation, so we can be sure the
+        # first uses the old, '.global' ID, and the second is the 'fixed' one
+        try:
+            global_entry = entries[0]
+            fixed_entry = entries[1]
+        except Exception as e:
```

Comment:
What exception would be getting here? `RangeError` or similar if there is no `fixed_entry`? Can't you just check or assert on `len(entries)` first?

Reply:
That's some leftovers from debugging the tests that, were breaking on Windows. Checking the length is definitely better.

I'll do some clean up once I get this test passing on Windows.